configure.ac

```diff
@@ -106,6 +106,8 @@ AC_FUNC_MMAP
 AC_FUNC_REALLOC
 AC_CHECK_FUNCS([clock_gettime ftruncate memmove memset strdup strstr strtoull strlen strchr])
 
+AX_GCC_FUNC_ATTRIBUTE(constructor)
```

Comment:
And sadly, we need one more test for `lttng` version to verify that they use constructor

Reply:
I will add it here. I will also update the `THAPI-spack` repo in a separate PR.